micro_sam/models/peft_sam.py

```diff
@@ -79,6 +122,10 @@ def __init__(
 
         assert rank > 0
 
+        dim = model.image_encoder.blocks[0].attn.qkv.in_features
+        self.FacTu = nn.Linear(dim, rank, bias=False)
```

Comment:
@anwai98 Do we need to add a condition here, so that these parameters are only initialized when FacT is used?

Reply:
Hmm, we definitely cannot hard-code them in the current location, would create a bit of a mess. While reviewing, Ill try to see how we can make it flexible upon the peft module.